### PR TITLE
Alien infestation random event now prompts ghosts about it

### PR DESCRIPTION
**Pull request by lzimann**
So it no longer kidnaps your ghost regardless if you want to play or not.
Also adds a never for this round option.

### CODE REVIEW

**Comment by coiax** [comment]
Adding a "never for this round" option for an event that can legitimately only fire once.

**Comment by lzimann** [comment]
Too late now.

**Comment by XDTM** [comment]
Any chance of this for normal larvas?

**Comment by coiax** [comment]
Yeah, I can do that, got something in mind as well.

File: code/modules/events/alien_infestation.dm [hunk 1]
```diff
@@ -50,7 +50,7 @@
 		message_admins("An event attempted to spawn an alien but no suitable vents were found. Shutting down.")
 		return MAP_ERROR
 
-	var/list/candidates = get_candidates("alien", null, ROLE_ALIEN)
+	var/list/candidates = pollCandidates("ADo you want to play as an alien larva?", ROLE_ALIEN, null, ROLE_ALIEN, 100, POLL_IGNORE_ALIEN_LARVA)
```
Comment (coiax) [inline code/modules/events/alien_infestation.dm]: this should be using the ghost_role subevent

Comment (RandomMarine) [inline code/modules/events/alien_infestation.dm]: Also, >Ado

Comment (lzimann) [inline code/modules/events/alien_infestation.dm]: What do you mean by ghost_role subevent?

Comment (coiax) [inline code/modules/events/alien_infestation.dm]: https://github.com/tgstation/tgstation/blob/master/code/modules/events/ghost_role.dm

**Comment by ExcessiveUseOfCobblestone** [comment]
> Adding a "never for this round" option for an event that can legitimately only fire once.

what did he mean by this?

**Comment by coiax** [comment]
The alien infestation event can only happen once via RNG. It's okay, I'm using the flag in #24523

**Comment by coiax** [comment]
This still needs to be a ghost_role subevent, like every other event that has ghost roles

**Comment by lzimann** [comment]
It already is a ghost_role.

File: code/modules/events/alien_infestation.dm [hunk 2]
```diff
@@ -50,7 +50,7 @@
 		message_admins("An event attempted to spawn an alien but no suitable vents were found. Shutting down.")
 		return MAP_ERROR
 
-	var/list/candidates = get_candidates("alien", null, ROLE_ALIEN)
+	var/list/candidates = pollCandidates("Do you want to play as an alien larva?", ROLE_ALIEN, null, ROLE_ALIEN, 100, POLL_IGNORE_ALIEN_LARVA)
```
Comment (coiax) [inline code/modules/events/alien_infestation.dm]: You need to use the get_candidates(jobban, gametypecheck, be_special) proc

Comment (lzimann) [inline code/modules/events/alien_infestation.dm]: Well then the proc is not working properly, since it was grabbing ghosts instead of asking.

Comment (coiax) [inline code/modules/events/alien_infestation.dm]: It works properly for all the other events that use it

**Comment by lzimann** [comment]
i'll just figure out why get_candidates won't work later

**Pull request closed without merging**